### PR TITLE
Fix: Enable proper append/update support for 3D scatter plots (`scatter3d`)

### LINKED ISSUE #702: I have a problem on 3D-scatter.
I don't know how to append points on the earlier 3D-scatter. I have tried when I type 2D tensors as the firsr scatter and then type 3D tensors, it can append the (x,y,z) points on the scatter. But it has 2D plot on the 3D plot. But if I type 3D tensor first, I can't append new 3D point.

**Here is my code**
```
import visdom
import numpy as np
import torch

viz = visdom.Visdom()
win = viz.scatter(
        X=torch.rand(5, 2), #when I change 2 to 3, it has problem
        opts=dict(
            markersize=10,
            # markercolor=np.random.randint(0, 5, (5, 3,)),
            showlegend=True,
        ),
    )

# 断言该窗口是否存在
assert viz.win_exists(win), 'Created window marked as not existing'

# 添加新的追踪到散点图中
x = torch.rand(5, 3)
print(x)
y = np.random.rand(5)
print(y)
# z = np.random.rand(5)
viz.scatter(
        X=x,
        #Y=y,
        win=win,
        name='new_trace',
        update='append',
        # opts=dict(showlegend=True)
)

viz.scatter(
    X=torch.rand(5, 3),
    # Y=np.random.rand(255),
    win=win,
    name='new_2',
    update='append'
)
```

**I don't how to append the new points on 3Dscatter,  Thank you!!!**

### PR DESCRIPTION
**Pull request by Manik-Khajuria-5**
## Description

This PR fixes an issue where updating an existing 3D scatter plot using `update='append'` was not working correctly. The fix ensures that `scatter3d` is correctly handled in the update flow, allowing consistent append behavior for 3D scatter plots.

---

## Motivation and Context
This resolves the issue described in:
## Fix :  https://github.com/fossasia/visdom/issues/702

Previously:
- Appending worked when starting with a 2D scatter plot and then adding 3D points (unexpected behavior)
- Appending failed when starting with a 3D scatter plot
- Error observed:

After this fix:
- 3D → 3D append works correctly
- Multiple sequential 3D updates work as expected
- Behavior is now consistent across 2D and 3D scatter plots

---

### Test 1 (Previously working case)
- Created a 2D scatter plot
- Appended 3D points  
Result: Works (unchanged behavior)

### Test 2 (Previously broken case)
- Created a 3D scatter plot
- Appended 3D points  
Result: Now works correctly

### Test 3 (Sequential updates)
- Created a 3D scatter plot
- Appended multiple 3D datasets sequentially  
Result: All updates work as expected

### Additional Testing
- Compared behavior before and after the fix
- Verified no regression in existing functionality
- Tested using demo-style scripts and manual runs

---

### Before : 

https://github.com/user-attachments/assets/629e18f7-9679-4cdb-91bb-6c90f51e53a4

---

### After : 

https://github.com/user-attachments/assets/3c67da83-431e-4fec-bc4b-ec0a4b2da02c

---

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor or cleanup (changes to existing code for improved readability or performance)

## Checklist:

- [x] I adapted the version number under `py/visdom/VERSION` according to [Semantic Versioning](https://semver.org/)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by Sourcery

Bug Fixes:
- Fix append/update handling so that existing scatter3d plots can be updated with additional 3D data without errors.

### CODE REVIEW

**Comment by Manik-Khajuria-5** [comment]
@vedansh-5 @Jayantparashar10  Can u please review this PR ?

**Review by tonypzy — APPROVED** [review]

File: py/visdom/server/handlers/web_handlers.py
Comment (rajnisht7) [inline py/visdom/server/handlers/web_handlers.py]: adding `scatter3d` to the allowed trace types makes sense, but i think the update logic also needs to handle the z axis for append operations

suggested changes:
```
axes = ["x", "y"]
            
if pdata[idx]["type"] == "scatter3d":
    axes.append("z")
for axis in axes:
    ....
```

Comment (Manik-Khajuria-5) [reply py/visdom/server/handlers/web_handlers.py]: @rajnisht7 Added Thanks for review

**Review by Jayantparashar10 — APPROVED** [review]
I have tested it locally and its working correctly. LGTM

Image

**Review by rajnisht7 — APPROVED** [review]

**Review by vedansh-5 — APPROVED** [review]
Image

Image

I have tested this locally, LGTM! Thanks @Manik-Khajuria-5

**Review by Saksham-Sirohi — APPROVED** [review]
